weather-app/src/components/FavoriteCities.js

```diff
@@ -0,0 +1,65 @@
+import Component from '../blackbox';
+import { addToArray, bindAll } from '../utils';
+
+class FavoriteCities extends Component {
+  constructor() {
+    super();
+
+    this.host.classList.add('favorite-list');
+
+    this.btn = document.createElement('button');
+    this.btn.type = 'button';
+    this.btn.title = 'Clear the list';
+    this.btn.classList.add('btn-delete');
+
+    bindAll(this, 'handleCityClick', 'handleDeleteClick');
+    this.host.addEventListener('click', this.handleCityClick);
+    this.btn.addEventListener('click', this.handleDeleteClick);
+  }
+
+  get list() {
+    return localStorage.getItem('favoriteCities') ? JSON.parse(localStorage.getItem('favoriteCities')) : [];
+  }
+
+  addToFavorite(location) {
+    const arr = addToArray(this.list, location);
+    localStorage.setItem('favoriteCities', JSON.stringify(arr));
+    this._render();
+  }
+
+  handleCityClick(ev) {
+    let target = ev.target;
+    while (target !== this.host) {
+      if (target.tagName === 'LI') {
+        const location = this.list[target.id];
+        this.props.onCityClick(location);
+        return;
+      };
+      target = target.parentNode;
```

Comment:
You don't need such complicated logic for this. Click event will bubble to the host and you can handle it there.

Something like:
```
if (ev.target.classList.contains('city-item`)) {
   this.props.onCityClick(this.list[ev.target.id]);
}
```

Reply:
And in other such situations to :)